File: hw/caizapanta/u3/HW23Observer/Observer/src/main/java/ec/edu/espe/observer/model/Investor.java
package ec.edu.espe.observer.model;

/**
 *
 * @author Caizapanta Tammy
 */
public class Investor implements IInvestor {

    private String name;

    public Investor(String name) {
        this.name = name;
    }

    @Override
    public void update(Stock stock, Object args) {
        System.out.println("Notified Observer : " + name);
        if (args instanceof String) {
            System.out.println("The symbol of : " + stock.getSymbol() + "Changed to: " + args);
        } else if (args instanceof Double) {
            System.out.println("The Price of : " + stock.getSymbol() + "Changed to: " + args);
        }
    }
}

File: hw/yanez/u3/HW23Observer/Observer/model/investor.py
# model/investor.py

class IInvestor:
    def update(self, stock, data):
        pass


class Investor(IInvestor):
    def __init__(self, name):
        self.name = name

    def update(self, stock, data):
        print(f"Notified observer {self.name}")
        if isinstance(data, str):
            print(f"The symbol of {stock.symbol} changed to: {data}")
        elif isinstance(data, float):
            print(f"The price of {stock.symbol} changed to: {data}")
